site/source/docs/optimizing/Optimizing-Code.rst

```diff
@@ -23,14 +23,15 @@ The optimization level you should use depends mostly on the current stage of dev
 
 - When first porting code, run *emcc* on your code using the default settings (without optimization). Check that your code works and :ref:`debug <Debugging>` and fix any issues before continuing.
 - Build with lower optimization levels during development for a shorter compile/test iteration cycle (``-O0`` or ``-O1``).
-- Build with ``-O2`` or ``-O3`` when releasing your code.  ``-O3`` builds are even more optimized than ``-O2``, but at the cost of significantly longer compilation time.
+- Build with ``-O2`` to get most optimizations.
+- Build with ``-O3`` or ``-Os`` when releasing your code. ``-O3`` builds are even more optimized than ``-O2``, but at the cost of significantly longer compilation time. ``-Os`` builds focus more on size than speed (and also can increase compilation times like ``-O3``).
 - Other optimizations are discussed in the following sections.
 
 In addition to the ``-Ox`` options, there are separate compiler options that can be used to control the JavaScript optimizer (:ref:`js-opts <emcc-js-opts>`), LLVM optimizations (:ref:`llvm-opts <emcc-llvm-opts>`) and LLVM link-time optimizations (:ref:`llvm-lto <emcc-llvm-lto>`).
 
 .. note::
 
-	-  The meanings of the *emcc* optimization flags (``-O1, -O2`` etc.) are different to the similarly-named options in *gcc*, *clang*, and other compilers, because optimizing JavaScript is very different to optimizing native code. The mapping of the *emcc* levels to the LLVM bitcode optimization levels is documented in the reference.
+	-  The meanings of the *emcc* optimization flags (``-O1, -O2`` etc.) are different to the similarly-named options in *gcc*, *clang*, and other compilers, because optimizing asm.js and WebAssembly is very different to optimizing native code. The mapping of the *emcc* levels to the LLVM bitcode optimization levels is documented in the reference.
```

Comment:
Is this really true? I'd guess that a large part of the optimization effects come from LLVM IR (where we pass the opt level directly through), and even outside of that we still try to make compile time/speed tradeoffs in a similar way, right? So while the correspondence between specific optimizations is not exact, I'd say the intent of the effects is close enough?

Reply:
It's generally similar, but the JS optimizations and metadce (JS/wasm optimization) things are pretty different from "normal" clang/gcc optimizations, that I think it's worth mentioning. But maybe this text is too scary?

Reply:
Maybe we could say that these flags control "additional" wasm- and js-specific optimizations (rather than "different")?